src/assets/netteForms.js

```diff
@@ -208,15 +207,60 @@ Nette.isDisabled = function(elem) {
 
 
 /**
- * Display error message.
+ * Adds error message to the queue.
  */
 Nette.addError = function(elem, message) {
-	if (message) {
-		alert(message);
+	Nette.errors.push({
+		elem: elem,
+		message: message
+	});
+};
+
+
+/**
+ * Display error messages.
+ */
+Nette.showErrors = function() {
+	var messages = '';
+	var focusElem;
+
+	for (var i in Nette.errors) {
+		var obj = Nette.errors[i];
+		var elem = obj.elem;
+		var message = obj.message;
+
+		if (!focusElem && elem.focus) {
+			focusElem = elem;
+		}
+
+		if (message) {
+			if (messages) {
+				messages += '\n';
+			}
+			messages += message;
+		}
 	}
-	if (elem.focus) {
-		elem.focus();
+
+	Nette.errors = [];
+	if (messages) {
+		Nette.alert(messages);
+
+		if (focusElem) {
+			focusElem.focus();
+		}
+
+		return false;
 	}
+
+	return true;
+};
+
+
+/**
+ * Shows an alert.
+ */
+Nette.alert = function(message) {
```

Comment:
I've added this while I'm working on "show all messages at once".

Tt would be nice to have an option redefine "show alert" function and create custom handlers.